packages/appeals-service-api/src/controllers/appeals-case-data.js

```diff
@@ -33,7 +34,15 @@ const getAppealByCaseRefAndLpaCode = async (req, res) => {
 	}
 };
 
+const postAppealCase = async (req, res) => {
```

Comment:
Think this function should have error handling and the service to throw predefined errors if we are to follow the pattern from elsewhere

Reply:
![image](https://github.com/Planning-Inspectorate/appeal-planning-decision/assets/7422644/358807ed-7b60-4299-b38d-6240879e7b6c)

Linter gives this error in the service 

```
Unnecessary try/catch wrapper.eslint[no-useless-catch](https://eslint.org/docs/rules/no-useless-catch)
```

Reply:
Can you think of any specific errors I should be trapping? The pattern seems to be used rather sporadically and I can't find a consistent example of this use case (adding a JSON object)

Reply:
@bryn-pins @suityou01 For now we have a requirement on the ticket to return a 400 if the payload does not contain an LPACode property as we are using this for sharding. Apart from that there are no other specific error handling scenarios so we shouldn't catch any exceptions, just let them bubble up and return as the default 500 response.

Reply:
So the only reason I mentioned it is because I believe the api handler in forms-web-app throws a json parsing error if the api doesn't respond in a specific format, which then isn't obvious in the frontend logs what that error was. i.e. a 500 status code with no json body will error frontend side

But happy to leave as is for now until we know if we need a friendly error for user

Reply:
Ah good shout if we're not handling this at a higher level. @bryn-pins @suityou01 We could make a middleware that we can re use on routes that require a payload of json content type instead of handling it verbosely on the controller than have the middleware return 400 if the payload type is invalid? Or do we already do something like that @bryn-pins ?

Reply:
If you mean api side to automatically send a json body in the event of an error that could work, although part of me just wants to redo the frontend (handler() in appeals-api-wrapper.js)

Either way would expect that in a different ticket than this one as it would impact all endpoints and I'm not sure I'd trust our test suite to catch any issue that came out of it which is why I've avoided it so far.

Reply:
Am going to resolve this thread for now